VMBackup/main/Utils/EventLoggerUtil.py

```diff
@@ -0,0 +1,256 @@
+    
+    
+    def __init__(self, log, event_directory, severity_level):
+        self.temporary_directory = os.path.join(event_directory, 'Temp')
+        self.space_available_in_event_directory = 0
+        self.event_processing_interval = 0
+        self.logger = log
+        self.disposed = False
+        self.event_processing_task = None  
+        self.current_message_len = 0
+        self.event_logging_enabled = False
+        self.event_logging_error_count = 0
+        self.events_folder = event_directory
+        self.event_logging_enabled = bool(self.events_folder)
+        self.filehelper = FileHelpers()
+
+        if self.event_logging_enabled:
+            self.extension_version = os.path.basename(os.getcwd())
+            self.operation_id = ''
+            self.log_severity_level = severity_level
+            print("Information: EventLogging severity level setting is ",self.log_severity_level)
+			# creating a temp directory
+            if not os.path.exists(self.temporary_directory):
+                os.makedirs(self.temporary_directory)
+            FileHelpers.clearOldJsonFilesInDirectory(self.temporary_directory)
+
+            FileHelpers.clearOldJsonFilesInDirectory(self.events_folder)
+            
+            self.event_processing_signal = threading.Event() # an event object that runs continuously until signal is set
+            self.current_message = ''
+            self.event_queue = queue.Queue()
+            
+			
+            space_available = LoggingConstants.MaxEventDirectorySize - FileHelpers.getSizeOfDir(self.events_folder)
+            self.space_available_in_event_directory = max(0, space_available)
+            print("Information: Space available in event directory : %sB" %(self.space_available_in_event_directory))
+            
+            self.event_processing_interval = LoggingConstants.MinEventProcesingInterval
+            print("Information: Setting event reporting interval to %ss" %(self.event_processing_interval))
+            
+            self.begin_event_queue_polling()
+        else:
+            print("Warning: EventsFolder parameter is empty. Guest Agent does not support event logging.")
+            
+    @staticmethod
+    def GetInstance(log, event_directory, severity_level):
+        if EventLogger._instance is None:
+            with EventLogger._lock:
+                if EventLogger._instance is None:
+
+                    EventLogger._instance = EventLogger(log, event_directory, severity_level)
+        return EventLogger._instance
+        
+    def update_properties(self, task_id):
+        self.operation_id = task_id
+
+    def trace_message(self, severity_level, message, *args):
+        level = 0
+        if (severity_level == "Verbose"):
+            level = 0
+        elif (severity_level == "Info"):
+            level = 1
+        elif (severity_level == "Warning"):
+            level = 2
+        elif(severity_level == "Error"):
+            level = 3
```

Comment:
Let's separate this section of code, into an enum outside of trace_message fn logic.